peps/pep-0750.rst

```diff
@@ -53,837 +33,1089 @@ deferred pending further experience with f-strings. Work on this PEP was
+`Examples`_ section for more information on how to process template strings.
 
-As with other string literal prefixes, no whitespace can be between the tag and the
-quote mark.
+The ``conv`` attribute is the `optional conversion <https://docs.python.org/3/library/string.html#format-string-syntax>`_
+to be used, one of ``r``, ``s``, and ``a``, corresponding to ``repr()``, 
+``str()``, and ``ascii()`` conversions. As with f-strings, no other conversions 
+are supported:
 
-PEP 701
--------
+.. code-block:: python
 
-Tag strings support the full syntax of :pep:`701` in that any string literal,
-with any quote mark, can be nested in the interpolation. This nesting includes
-of course tag strings.
+    name = "World"
+    template = t"Hello {name!r}"
+    assert template.args[1].conv == "r"
 
-Evaluating Tag Strings
-----------------------
+If no conversion is provided, ``conv`` is ``None``. 
 
-When the tag string is evaluated, the tag must have a binding, or a ``NameError``
-is raised; and it must be a callable, or a ``TypeError`` is raised. The callable
-must accept a sequence of positional arguments. This behavior follows from the
-de-sugaring of:
+The ``format_spec`` attribute is the `format specification <https://docs.python.org/3/library/string.html#format-specification-mini-language>`_.
+As with f-strings, this is an arbitrary string that defines how to present the value:
 
 .. code-block:: python
 
-    trade = 'shrubberies'
-    mytag'Did you say "{trade}"?'
+    value = 42
+    template = t"Value: {value:.2f}"
+    assert template.args[1].format_spec == ".2f"
 
-to:
+Format specifications in f-strings can themselves contain interpolations. This
+is permitted in template strings as well; ``format_spec`` is set to the eagerly 
+evaluated result:
 
 .. code-block:: python
+   
+    value = 42
+    precision = 2
+    template = t"Value: {value:.{precision}f}"
+    assert template.args[1].format_spec == ".2f"
 
-    mytag(DecodedConcrete(r'Did you say "'), InterpolationConcrete(lambda: trade, 'trade', None, None), DecodedConcrete(r'"?'))
-
-.. note::
+If no format specification is provided, ``format_spec`` defaults to an empty 
+string (``""``). This matches the ``format_spec`` parameter of Python's 
+`format() <https://docs.python.org/3/library/functions.html#format>`_ built-in.
 
-    `DecodedConcrete` and `InterpolationConcrete` are just example implementations. If approved,
-    tag strings will have concrete types in `builtins`.
+Unlike f-strings, it is up to code that processes the template to determine how to
+interpret the ``conv`` and ``format_spec`` attributes. It is not required that they
+be used in any particular way, but it is recommended that they be respected when
+present and, to the extent possible, match the behavior of f-strings. It would
+be surprising if, for example, a template string that uses ``{value:.2f}`` did not
+round the value to two decimal places when processed.
```

Comment:
Perhaps rewrite in the active voice?:

```suggestion
Unlike f-strings, it is up to code that processes the template to determine how to
interpret the ``conv`` and ``format_spec`` attributes. 
Such code is not required to use these attributes, but when present they should
be respected, and to the extent possible match the behavior of f-strings.
It would be surprising if, for example, a template string that uses ``{value:.2f}``
did not round the value to two decimal places when processed.
```